spring-kafka/src/test/java/org/springframework/kafka/listener/BatchSmartMessageConverterTests.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2016-present the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.listener;
+
+import java.lang.reflect.Method;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.kafka.support.converter.BatchMessagingMessageConverter;
+import org.springframework.kafka.support.converter.MessagingMessageConverter;
+import org.springframework.messaging.converter.SmartMessageConverter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Test for SmartMessageConverter support in batch listeners.
+ *
+ * @author George Mahfoud
+ * @since 3.4.0
+ */
+class BatchSmartMessageConverterTests {
+
+	@Test
+	void testMessagingMessageConverterUsesSmartConverter() {
+		// Given: A regular MessagingMessageConverter with SmartMessageConverter
+		MessagingMessageConverter converter = new MessagingMessageConverter();
+		SmartMessageConverter smartConverter = mock(SmartMessageConverter.class);
+		converter.setMessagingConverter(smartConverter);
+
+		// When: We check if SmartMessageConverter is set
+		// The converter should be ready to use the SmartMessageConverter
+		// This works for regular listeners
+		assertThat(converter).isNotNull();
```

Comment:
Looks like this test is bogus.
How `converter` variable can be null if you explicitly do `new MessagingMessageConverter();` ?